apps/accounts/tasks/vault.py

```diff
@@ -52,7 +52,8 @@ def sync_secret_to_vault():
         for model in to_sync_models:
             instances += list(model.objects.all())
 
-        with ThreadPoolExecutor(max_workers=10) as executor:
+        max_workers = 1 if VaultTypeChoices.azure == vault_client.type else max_workers = 10
+        with ThreadPoolExecutor(max_workers=max_workers) as executor:
             tasks = [executor.submit(sync_instance, instance) for instance in instances]
 
             for future in as_completed(tasks):
```

Comment:
It looks like there might be a slight typo in your code snippet:

There is no newline between `with ThreadPoolExecutor(max_workers=10)` and the next line.

Here's how it should look based on your comments:

```python
def sync_secret_to_vault():
    # Code below remains unchanged
```

If you meant that these two statements were intended before `if __name__ == '__main__':` (the condition where this block runs), then there doesn't seem to be an irregularity or issue. The current logic seems correct for async operations, but you could optimize further by removing unused variables if necessary for cleaner code formatting:

```python
async def fetch_and_update_instance(model):
    ...
    
# Fetch all models
to_sync_models = list(models)

sync_instances(to_sync_models)
```
This reduces unnecessary variable reassignment within the loop, aiming at keeping the script concise while ensuring clarity and efficiency without compromising performance. Also note, I removed `ThreadPoolExecutor()` since we now have separate asynchronous functions (`fetch_and_update_instance`) which encapsulate the core functionality better.